server/sonar-server/src/main/java/org/sonar/server/source/index/SourceLineIndex.java

```diff
@@ -112,4 +113,20 @@ public SourceLineDoc getLine(String fileUuid, int line) {
     }
     throw new NotFoundException(String.format("No source found on line %s for file '%s'", line, fileUuid));
   }
+
+  public Date lastCommitedDateOnProject(String projectUuid) {
+    SearchRequestBuilder request = getClient().prepareSearch(SourceLineIndexDefinition.INDEX)
+      .setTypes(SourceLineIndexDefinition.TYPE)
+      .setSize(1)
+      .setQuery(QueryBuilders.boolQuery()
```

Comment:
Specify the field to return with setFetchSource(new String[]{SourceLineIndexDefinition.FIELD_SCM_REVISION}) to decrease memory used by the query